corehq/apps/translations/app_translations/upload_app.py

```diff
@@ -96,9 +96,12 @@ def get_expected_headers(sheet_name):
     if error:
         msgs.append((messages.error, error))
         return msgs
-
-    expected_headers_by_sheet_name = {k: v for k, v in get_bulk_app_sheet_headers(app, lang=lang)}
-    expected_headers_by_id = {k: v for k, v in get_bulk_app_sheet_headers(app, lang=lang, by_id=True)}
+    # if there is a lang selected, assume that user submitted a single sheet
+    single_sheet = bool(lang)
```

Comment:
There's an `is_single_sheet` function based on sheet name used down on line 118/121, I'd use that here.

Reply:
That happens within the sheets iteration, i would need to do something like 
`is_single_sheet(workbook.worksheets[0].title)` here which sounds fine for now but brittle.